### PR TITLE
Consolidate cudf object handling in as_column

### PR DESCRIPTION
**Pull request by mroeschke**
## Description
Since these paths are very similar, consolidating these paths and returning early for this case

## Checklist
- [ ] I am familiar with the [Contributing Guidelines](https://github.com/rapidsai/cudf/blob/HEAD/CONTRIBUTING.md).
- [ ] New or existing tests cover these changes.
- [ ] The documentation is up to date with these changes.

### CODE REVIEW

File: python/cudf/cudf/core/column/column.py
Comment (vyasr) [inline python/cudf/cudf/core/column/column.py:1923]:
```suggestion
            return column.astype(dtype)
```

File: python/cudf/cudf/core/column/column.py
Comment (vyasr) [inline python/cudf/cudf/core/column/column.py:1920]: You could remove the else clause if you reused the name `arbitrary`. Up to you if you like that, just a suggestion.

File: python/cudf/cudf/core/column/column.py
Comment (vyasr) [inline python/cudf/cudf/core/column/column.py:1921]: Good question. I think ignoring it is probably right since that flag is primarily designed to help with ingestion from sources that don't properly support differentiating NaN and null (aka pandas, especially at the time this code was written). I'll defer to @shwina in case he has different thoughts though.

**Review by vyasr — APPROVED** [review]
A couple suggestions, but nothing blocking. Would wait for @shwina's feedback on the `nan_as_null` question.

File: python/cudf/cudf/core/column/column.py
Comment (shwina) [reply python/cudf/cudf/core/column/column.py:1921]: Yeah I think that's right. Although it may then be helpful to add something like this in the docstring:

```
nan_as_null: bool
    Use N/A in the output where there are NaNs in the input. This parameter only applies when arbitrary is a <...>. 
```

File: python/cudf/cudf/core/column/column.py
Comment (mroeschke) [reply python/cudf/cudf/core/column/column.py]: Sure thing. Changed

File: python/cudf/cudf/core/column/column.py
Comment (mroeschke) [reply python/cudf/cudf/core/column/column.py]: Sure thing. Changed

File: python/cudf/cudf/core/column/column.py
Comment (mroeschke) [reply python/cudf/cudf/core/column/column.py]: Thanks for confirming. Updated the docstring.

**Pull request merged**
